core/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -166,7 +187,7 @@ public int hashCode() {
     @Override
     public IndexTemplateMetaData readFrom(StreamInput in) throws IOException {
         Builder builder = new Builder(in.readString());
-        builder.order(in.readInt());
+        builder.order(in.readVInt());
```

Comment:
Agreed. I had a moment of insanity and forgot that 2.x cluster states won't be using that.